From 8fe6ee5ae58e1fba9837fac92e1c6927d550796b Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Tue, 1 Mar 2016 11:51:13 +0100
Subject: [PATCH virt-viewer] ovirt: Only use active ISO domains for foreign
 menu

oVirt storage domains can be in various states (inactive, in
maintainance, ...). We only want to show the ISOs it contains in the
foreign menu when the storage domain is actually active, not in the
other states.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1313376
---
 src/ovirt-foreign-menu.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/ovirt-foreign-menu.c b/src/ovirt-foreign-menu.c
index 0cb2fda..49f6843 100644
--- a/src/ovirt-foreign-menu.c
+++ b/src/ovirt-foreign-menu.c
@@ -651,12 +651,17 @@ static void storage_domains_fetched_cb(GObject *source_object,
     while (g_hash_table_iter_next(&iter, NULL, (gpointer *)&domain)) {
         OvirtCollection *file_collection;
         int type;
+        int state;
 
-        g_object_get(domain, "type", &type, NULL);
+        g_object_get(domain, "type", &type, "state", &state, NULL);
         if (type != OVIRT_STORAGE_DOMAIN_TYPE_ISO) {
             continue;
         }
 
+        if (state != OVIRT_STORAGE_DOMAIN_STATE_ACTIVE) {
+            continue;
+        }
+
         file_collection = ovirt_storage_domain_get_files(domain);
         if (file_collection != NULL) {
             if (menu->priv->files) {
